numpy/_core/tests/test_einsum.py

```diff
@@ -1282,3 +1282,29 @@ def test_overlap():
     #gh-10080, out overlaps one of the operands
     c = np.einsum('ij,jk->ik', a, b, out=b)
     assert_equal(c, d)
+
+def test_einsum_chunking_precision():
+    """Most einsum operations are reductions and until NumPy 2.3 reductions
+    never (or almost never?) used the `GROWINNER` mechanism to increase the
+    inner loop size when no buffers are needed.
+    Because einsum reductions work roughly:
+
+        def inner(*inputs, out):
+            accumulate = 0
+            for vals in zip(*inputs):
+                accumulate += prod(vals)
+            out[0] += accumulate
+
+    Calling the inner-loop more often actually improves accuracy slightly
+    (same effect as pairwise summation but much less).
+    Without adding pairwise summation to the inner-loop it seems best to just
+    not use GROWINNER, a quick tests suggest that is maybe 1% slowdown for
+    the simplest `einsum("i,i->i", x, x)` case.
+
+    (It is not clear that we should guarantee precision to this extend.)
+    """
+    num = 100000000
+    value = 1.00000000002
+    res = np.einsum("i->", np.full(num, value)) / num
```

Comment:
I'd tend to use `float32` and fewer elements, just to keep the test fast (now 147 ms on my machine).

Reply:
Doesn't work :).  Einsum uses double for the accumulation, so for float32, the results are actually better without chunking...

But, let me use broadcasting and make the number smaller, certainly doesn't need this much.